client/extensions/hello-dolly/state/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import {
+	ROUTE_SET,
+	SECTION_SET,
+	SITE_SETTINGS_SAVE,
+} from 'state/action-types.js';
+
+export default function( state = 0, action ) {
+	switch ( action.type ) {
+		case ROUTE_SET:
+		case SECTION_SET:
+		case SITE_SETTINGS_SAVE:
+			return advanceToNextLyric( state );
+		default:
+			return state;
+	}
+}
+
+function advanceToNextLyric( index ) {
+	return index + 1;
+}
```

Comment:
fair enough. with _such simple logic_ it's harder for me to see it this way, but make the logic just a little more complicated and I'm 💯 with you.